log4j-core/src/main/java/org/apache/logging/log4j/core/appender/SyslogAppender.java

```diff
@@ -104,6 +104,9 @@
         @PluginElement("LoggerFields")
         private LoggerFields[] loggerFields;
 
+        @PluginBuilderAttribute( "timestampPrecision" )
+        private String timestampPrecision;
```

Comment:
Shouldn't we use an enum here?

Reply:
I agree. An enum would make sense.

Reply:
I have converted it into an enum with latest commit.